hamilton/driver.py

```diff
@@ -1807,6 +1828,39 @@ def with_materializers(
         self.materializers.extend(materializers)
         return self
 
+    def with_cache(
+        self,
+        path: Union[str, pathlib.Path] = "hamilton_cache",
```

Comment:
Local subdirectory is a safer default IMO. The user *sees* the data in his filesystem and it becomes easier to delete files to clear disk space. Setting a cache per project seems like a reasonable pattern.

Once we have better cache eviction and storage management mechanisms, we could use a global cache `~/.hamilton`. We could use an environment variable to toggle between the two.

Reply:
I guess? Seems very error-prone to me -- easy to accidentally invalidate by running it from the wrong place. It's generally bad (IMO) to rely on the CWD for program behavior, especially if it's a hard-to-debug side-effect. But if this is too much to change let's keep it like this for now.

Reply:
That's how MLFlow works by default, so it would feel familiar. It wouldn't be difficult to change the default location, but having a single central storage would require more utility for disk space management

Reply:
Changing this string is sufficient to modify the default location in `.with_cache()`,  which would affect the default location of the `metadata_store` and the `result_store`. I can make the change tomorrow

Reply:
Yeah I'm happy with whatever you feel is the best user experience, let's just make it clear in the docs